vars/influxWriteData.groovy

```diff
@@ -0,0 +1,64 @@
+import com.sap.piper.ConfigurationLoader
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.JsonUtils
+
+def call(Map parameters = [:]) {
+
+    def stepName = 'influxWriteData'
+
+    handlePipelineStepErrors (stepName: stepName, stepParameters: parameters, allowBuildFailure: true) {
+
+        def script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        prepareDefaultValues script: script
+
+        final Map stepDefaults = ConfigurationLoader.defaultStepConfiguration(script, stepName)
+        final Map stepConfiguration = ConfigurationLoader.stepConfiguration(script, stepName)
+        final Map generalConfiguration = ConfigurationLoader.generalConfiguration(script)
+
+        List parameterKeys = [
+            'artifactVersion',
+            'influxServer',
+            'influxPrefix'
+        ]
+        Map pipelineDataMap = [
+            artifactVersion: commonPipelineEnvironment.getArtifactVersion()
+        ]
+        List stepConfigurationKeys = [
+            'influxServer',
+            'influxPrefix'
+        ]
+
+        Map configuration = ConfigurationMerger.mergeWithPipelineData(parameters, parameterKeys, pipelineDataMap, stepConfiguration, stepConfigurationKeys, stepDefaults)
```

Comment:
IMHO `commonPipelineEnvironment` should keep all data we gather during a pipeline run. This is not configuration data but data collected during the run.

There are steps where we want to retrieve the values set during the pipeline run.
Nonetheless, we should not assume that a value exists otherwise the steps will not be self-contained anymore.
Thus, we should allow to mix and match step parameters with the values collected during the pipeline run.

I don't consider the data collected = general configuration and therefore I would prefer to tackle them separately.